### ISSUE TITLE
Incremental search skipping matches while jumping to next/previous match

### ISSUE THREAD

**Issue by purpleP** [neovim/neovim#7978]
```
NVIM v0.2.2
Build type: Debug
LuaJIT 2.0.5
Compilation: /usr/bin/cc -Wconversion -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -I/tmp/neovim-0.2.2/build/config -I/tmp/neovim-0.2.2/src -I/tmp/neovim-0.2.2/.deps/usr/include -I/tmp/neovim-0.2.2/.deps/usr/include -I/tmp/neovim-0.2.2/.deps/usr/include -I/tmp/neovim-0.2.2/.deps/usr/include -I/tmp/neovim-0.2.2/.deps/usr/include -I/tmp/neovim-0.2.2/.deps/usr/include -I/usr/include -I/tmp/neovim-0.2.2/build/src/nvim/auto -I/tmp/neovim-0.2.2/build/include

Features: +acl +iconv +jemalloc +tui 
See ":help feature-compile"
```
OS - ubuntu 17.10
Terminal - Konsole
```
echo $TERM
xterm-256color
```
File content

```
a = (I(1, 2), (3, 4), (5, 6))
# not a content   skip      ^
a = ((1, 2), (3, 4), (5, 6), (7, 8), (9, 10))
#         ^
```
`I` is the cursor.

If I start searching for `)` from this point and then start jumping to next matches while searching with `C-g/t` neovim skips last `)` on the first line and jumps to the first `)` on the second line. Which is obviously wrong. And vim is actually doing the right thing.

**Comment by KillTheMule** [comment]
Tried to reproduce, but your steps aren't fully understandable for me, could you rephrase please? I assume below your cursor there's a whitespace?

**Comment by purpleP** [comment]
@KillTheMule lines started with `#` aren't the content of example file they just to my comments. Example file has two lines in it. When jumping to next match in incremental search mode neovim skips latest matching `)` on the first line.

**Comment by KillTheMule** [comment]
I could not reproduce that on 2.3-580-gec62f925 (pretty recent) on windows with default settings. Both `C-g` and `C-t` do not skip that last matching `)` on the first line. Did you try with `nvim -u NONE`?

**Comment by purpleP** [comment]
@KillTheMule Yes I did.

**Comment by KillTheMule** [comment]
There were some patches related to that functionality as of late, maybe you could try the development version.

**Comment by purpleP** [comment]
@KillTheMule Thanks to appimage of the dev version I can say that dev version is working properly.
